File: 백준/Silver/1764. 듣보잡/듣보잡.py
from sys import stdin

n, m = map(int, stdin.readline().split())
nh = [stdin.readline().strip() for _ in range(n)]
ns = [stdin.readline().strip() for _ in range(m)]

nsh = sorted(list(set(nh) & set(ns)))

print(len(nsh))
for i in nsh:
    print(i)

File: 백준/Silver/1764. 듣보잡/듣보잡.java
import java.util.*;
import java.io.*;

class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int[] nm = Arrays.stream(br.readLine().split(" "))
				.mapToInt(Integer::parseInt).toArray();
		Set<String> set1 = new HashSet<>();
		Set<String> set2 = new HashSet<>();
		for (int i = 0; i < nm[0]; i++) set1.add(br.readLine());
		for (int i = 0; i < nm[1]; i++) set2.add(br.readLine());
		set1.retainAll(set2);
		System.out.println(set1.size());
		set1.stream().sorted().forEach(System.out::println);
	}
}